src/Http/Retry.php

```diff
@@ -0,0 +1,56 @@
+<?php
+namespace ChartMogul\Http;
+
+use Psr\Http\Message\ResponseInterface;
+use STS\Backoff\Backoff;
+
+class ExponentialStrategy extends \STS\Backoff\Strategies\ExponentialStrategy
+{
+    // min wait time
+    protected $base = 1000;
+}
+
+class Retry
+{
+
+    private $retries;
+
+    public function __construct($retries)
+    {
+        $this->retries = $retries;
+    }
+
+    private function retryHTTPStatus($status)
+    {
+        return $status == 429 || ($status >= 500 && $status < 600);
+    }
+
+    protected function shouldRetry($attempt, $maxAttempts, ResponseInterface $response = null, \Exception $ex = null)
+    {
+        if ($attempt >= $maxAttempts && ! is_null($ex)) {
```

Comment:
What happens if this is true `$attempt >= $maxAttempts`, meaning it reached the maximum number of retries, but this is false `!is_null($ex)`, meaning there was no exception thrown? Is doesn't look like it will stop.
Shouldn't this check be with `||` instead of `&&`?

Reply:
I would actually like to know the answer to @bogdanguranda question

Reply:
So if there is no exception and the HTTP status isn't 429/5xx, then the function would return null (falsy value) and we would never retry. So `$attempt` would never exceed `$maxAttempts`.

Reply:
Ok. And it will actually get here:
```
        if (!is_null($response)) {
            return $attempt < $maxAttempts && $this->retryHTTPStatus($response->getStatusCode());
        }
```
And this will return `false` which is correct, as attempts exceeded max attempts.